### DISCUSSION TITLE
dble支持mgr吗？

### DISCUSSION THREAD

**Discussion by Dinosauria** [actiontech/dble#3184] [Q&A] [answered]
dble支持mgr吗？

**Comment by PanternBao** [comment, marked as answer]
[https://actiontech.github.io/dble-docs-cn/1.config_file/1.03_user.xml.html](https://actiontech.github.io/dble-docs-cn/1.config_file/1.03_user.xml.html) 这里是配置说明，需要注意：dble内部逻辑没有对mgr做单独处理，因为mgr没有原生提供切换时候的对外的主动通知，需要通过订阅来主动发现，所以对外部组件(dble)来说，和其他半同步等高可用方式没有太大区别。目前可以通过自定义脚本对mgr切换做响应，dble 提供了单实例部署时候用于切换的python脚本[https://actiontech.github.io/dble-docs-cn/2.Function/2.12_failover.html](https://actiontech.github.io/dble-docs-cn/2.Function/2.12_failover.html)

但这部分仅用于demo，不建议直接上生产，生产需要更健壮的高可用组件

**Reply by Dinosauria** [reply]
在使用MGR时，需要注意避免使用 show slave status作为心跳语句，原因为这个sql在组复制模式下返回值为空，不能用来判断group-member复制状态正常与否。若在MGR模式下配置了show slave status作为心跳语句会出现如下影响：
+ [启动dble之后，日志会一直报" found MySQL master/slave Replication err](https://github.com/actiontech/dble/discussions/3104)
+ 当启动了[读操作的负载均衡rwSplitMode](https://actiontech.github.io/dble-docs-cn/1.config_file/1.04_db.xml.html)为1/2/3时，读操作不能均衡到slave（其中1会有明显的报错，但配置为2,3时，读sql会默默发送到master，起不到读操作的负载均衡作用）

**Reply by kevinbin** [reply]
dble支持MGR下的复制延迟检测吗

**Reply by PanternBao** [reply]
> dble支持MGR下的复制延迟检测吗

暂不支持

**Reply by ylinzhu** [reply]
从3.22.11.0开始，dble增加新的延迟检测模式，可以在MGR时使用
